File: mobile/src/types/library.ts
export type OccupancyStatus = 'Low' | 'Medium' | 'High';

export interface Library {
  id: string;
  name: string;
  address: string;
  city: string;
  district: string;
  capacity: number;
  currentOccupancy: number;
  occupancyPercentage: number;
  occupancyStatus: OccupancyStatus;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}

export interface LibraryQueryParams {
  search?: string;
  city?: string;
  district?: string;
  pageNumber?: number;
  pageSize?: number;
  sortBy?: 'name' | 'occupancy';
  sortDescending?: boolean;
}

export interface CheckInRequest {
  qrToken: string;
}

export interface CheckOutRequest {
  qrToken: string;
}

export type OccupancyLogType = 'CheckIn' | 'CheckOut';

export interface CheckInOutResult {
  libraryId: string;
  currentOccupancy: number;
  capacity: number;
  occupancyPercentage: number;
  occupancyStatus: OccupancyStatus;
  type: OccupancyLogType;
  timestamp: string;
}

export interface MyCheckInStatus {
  libraryId: string | null;
  libraryName: string | null;
  checkedInAt: string | null;
}

File: backend/LibraryOccupancy.Api/Models/Library.cs
namespace LibraryOccupancy.Api.Models;

public class Library : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string District { get; set; } = string.Empty;
    public int Capacity { get; set; }
    public int CurrentOccupancy { get; set; }
    public string QrCodeToken { get; set; } = string.Empty;

    public ICollection<OccupancyLog> OccupancyLogs { get; set; } = new List<OccupancyLog>();
}
